tests/test_integrated_channels/test_blackboard/test_client.py

```diff
@@ -77,25 +83,93 @@ def test_course_completion_with_no_course(self):
+        client._patch = unittest.mock.MagicMock(  # pylint: disable=protected-access
+            name="_patch",
+            return_value=success_response
+        )
+        client._resolve_blackboard_course_id = unittest.mock.MagicMock(  # pylint: disable=protected-access
+            name="_resolve_blackboard_course_id",
+            return_value="a-course-id"
+        )
+
+        status_code, status_text = client.update_content_metadata(serialized_data)
+
+        assert status_code == 200
+        assert status_text == "hooray"
+        assert client._patch.called  # pylint: disable=protected-access
+        assert client._resolve_blackboard_course_id.called  # pylint: disable=protected-access
+
+    def test_delete_content_metadata(self):
+        client = self._create_new_mock_client()
+        serialized_data = json.dumps({
+            "externalId": "a-course-id"
+        }).encode('utf-8')
+        success_response = unittest.mock.Mock(spec=Response)
+        success_response.status_code = 202
+        success_response.text = ""
+        success_response.json.return_value = {}
+
+        client._create_session()  # pylint: disable=protected-access
+        client._delete = unittest.mock.MagicMock(  # pylint: disable=protected-access
+            name="_delete",
+            return_value=success_response
+        )
+        client._resolve_blackboard_course_id = unittest.mock.MagicMock(  # pylint: disable=protected-access
+            name="_resolve_blackboard_course_id",
+            return_value="a-course-id"
+        )
+
+        status_code, status_text = client.delete_content_metadata(serialized_data)
+
+        assert status_code == 202
+        assert status_text == ""
+        assert client._delete.called  # pylint: disable=protected-access
+        assert client._resolve_blackboard_course_id.called  # pylint: disable=protected-access
+
     def test_client_behavior_on_successful_learner_data_transmission(self):  # pylint: disable=protected-access
         """
         Test that given successful requests for Blackboard learner data,
         the client makes the appropriate _patch call to update a user's grade
         """
-        client = BlackboardAPIClient(self.enterprise_config)
-        # Mock the auth process
-        client._get_oauth_access_token = unittest.mock.MagicMock(  # pylint: disable=protected-access
-            name="_get_oauth_access_token",
-            return_value=(self.token, 10000)
-        )
+        client = self._create_new_mock_client()
 
         # Mock the course ID request
-        client._get_courses_by_internal_id = unittest.mock.MagicMock(  # pylint: disable=protected-access
-            name="_get_courses_by_internal_id",
-            return_value=[{
-                'externalId': self.course_id,
-                'id': self.blackboard_course_id
-            }]
+        client._resolve_blackboard_course_id = unittest.mock.MagicMock(  # pylint: disable=protected-access
```

Comment:
refactored client a bit so we enclose the fetching of blackboard course id into one function, hence this change